lib/router/router.js

```diff
@@ -1,14 +1,14 @@
 /* global __NEXT_DATA__ */
 
-import { parse, format } from 'url'
+import { parse, format, resolve as resolveURL } from 'url'
 import evalScript from '../eval-script'
 import shallowEquals from '../shallow-equals'
 import { EventEmitter } from 'events'
 import { reloadIfPrefetched } from '../prefetch'
 import { loadGetInitialProps } from '../utils'
 
 export default class Router extends EventEmitter {
-  constructor (pathname, query, { Component, ErrorComponent, err } = {}) {
+  constructor (pathname, query, { Component, ErrorComponent, err, assetPrefix } = {}) {
```

Comment:
Do we want it in the "opts" object or as a direct param?

```js
constructor (pathname, query, { Component, ErrorComponent, err, assetPrefix } = {}) {
```

vs

```
constructor (pathname, query, assetPrefix, { Component, ErrorComponent, err } = {}) {
```

I'm not sure which is cleaner, but happy to refactor if you like.